rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -425,16 +435,7 @@ private void resolveParentPropertiesAndRepositoriesRecursively(List<Pom> pomAnce
 
             if (pom.getParent() != null) {
                 Pom parentPom = resolveParentPom(pom);
-
-                for (Pom ancestor : pomAncestry) {
```

Comment:
Why do we have to re-resolve every parent now? To collect all values upstream?

Reply:
Actually this is an another improvement I accidentally added in this PR. But no bother, because it is better. To answer your question we do already re-resolve every parent, I just remove the unnecessary List wrapper.